### PR TITLE
Modernized the packaging / testing configuration

### PR DESCRIPTION
**Pull request by agronholm**
* Moved metadata to setup.cfg, keeping just a stub setup.py file for editable installs (needed for the time being)
* Switched from Travis to Github Actions
* Switched from unittest to pytest
* Added tox configuration for easy local testing across different Python versions

### CODE REVIEW

**Comment by agronholm** [comment]
I don't know why the workflow was not triggered.

File: tests/test_tzlocal.py
Comment (pganssle) [inline tests/test_tzlocal.py:24]: While passing paths through the `tz_path` fixture via indirect parameterization is clever, it's quite obscure and it's not at all clear what this does without reading the documentation and/or knowing some pretty arcane `pytest` lore.

How about just refactoring this into a function that you call, or refactoring the tests into a class (not a `TestCase`) with a method like `self.path`?

**Review by pganssle — CHANGES_REQUESTED** [review]

**Comment by pganssle** [comment]
> I don't know why the workflow was not triggered.

I suspect that you cannot add new workflows that are triggered unless you have write access to the repository? I don't know the exact rules, but it's common to restrict this sort of thing so that people don't open thousands of PRs that run bitcoin miners or whatever on random unmaintained OSS projects.

File: tox.ini
Comment (ofek) [inline tox.ini:3]:
```suggestion
envlist = py{36,37,38,39}
```

File: .github/workflows/test.yml
Comment (agronholm) [reply .github/workflows/test.yml:36]: I'm aware, but in my own projects I'm not running tox in CI because of the significant additional overhead from virtualenv creation.

File: setup.cfg
Comment (agronholm) [reply setup.cfg:9]: Oops, yeah.

File: setup.cfg
Comment (agronholm) [reply setup.cfg:13]: I find keywords pretty pointless anyway. I'll remove pytz.

File: setup.cfg
Comment (agronholm) [reply setup.cfg:34]: I'll change that.

File: tests/test_tzlocal.py
Comment (agronholm) [reply tests/test_tzlocal.py:1]: That was purely a mistake. I'll fix that.

File: tests/test_tzlocal.py
Comment (agronholm) [reply tests/test_tzlocal.py]: Refactored into a plain function.

File: .github/workflows/test.yml
Comment (agronholm) [reply .github/workflows/test.yml:36]: I could switch this to use the src-layout if @regebro is willing.

**Comment by agronholm** [comment]
@regebro do you want to move tzlocal to use the `src` layout? I can modify this PR accordingly.

**Comment by regebro** [comment]
No, I prefer this layout, unless there is some new reason against it.

File: .github/workflows/test.yml
Comment (regebro) [reply .github/workflows/test.yml:36]: tox has for me always been more work than it's worth, so much so I even wrote my own tool instead, but the rise of Travis and other tools means I generally don't use either now. I run the tests locally for one version, and rely on the Github integration to run the others.

So I don't feel the need for a tox.ini at all, but others may disagree.

**Comment by agronholm** [comment]
The reason is that running the tests in the project directory will import the project source rather than the installed version of the package. That in turn means that the test suite won't expose any mishaps in the packaging (like missing files in `MANIFEST.in`). Moving the source to `src/` prevents that from happening. I've switched to the `src` layout in all my recent projects and am in the process of migrating older projects to that as well as I update them.

File: pyproject.toml
Comment (regebro) [inline pyproject.toml:1]: Well, learning about PIP518 was a disappointment. Ah well, nothing we can do about that now, I should have followed the PEP discussions in 2016. My bad.

**Review by regebro — APPROVED** [review]
Looks good, although we could possibly drop the tox.ini. Waiting for feedback on that before merge.

File: pyproject.toml
Comment (agronholm) [reply pyproject.toml:1]: I use tox routinely to run linting checks (which admittedly you have none) and then the test suite on several Python versions. I find that this saves me a lot of time compared to just waiting for the CI to run which is not that fast. But it's your project and your call, so do I remove tox.ini?

Comment (regebro) [reply pyproject.toml:1]: No, we can keep it. I'll think about the src/ change, but we can do it separately.

"The reason is that running the tests in the project directory will import the project source rather than the installed version of the package."

Mmm, yeah, see that's what I like about it. :-)

"(like missing files in MANIFEST.in)."

Ah, I've always used check-manifest for that.

**Comment by agronholm** [comment]
Looks like the workflow is running now, and failing on Windows.

**Comment by regebro** [comment]
Yeah, I wonder what the windows time zone is in github actions.

**Comment by agronholm** [comment]
Well, the traceback says `Etc/GMT`. Go figure.

**Comment by agronholm** [comment]
Oh, could this be caused by the missing `tzdata`?

**Comment by agronholm** [comment]
I added tzdata in #100, but that is now failing with a different error: https://github.com/regebro/tzlocal/pull/100/checks?check_run_id=1176719892

**Comment by agronholm** [comment]
Oh, it looks like a unix specific test. It should probably be skipped on Windows, yes?

**Pull request merged**
